test/jit/test_tracer.py

```diff
@@ -1407,6 +1406,36 @@ def forward(self, x):
 
         traced = torch.jit.trace(Wrapper(), (torch.rand(3, 4),))
 
+    def test_trace_multi_output_function(self):
```

Comment:
Does this test fail without shape information on original call note?

Reply:
It doesn't but it still checks the newly added part. If new change is wrong (I tried randomly assigned some shapes), this test may fail.

Reply:
In that case, could you use `FileCheck()` to ensure correct tensor information is in the IR? That would make the test fail in case of tensor properties being propagated differently

Reply:
I have added new logics to directly examine my change:
```
        for n in traced.graph.nodes():
            if n.kind() != 'prim::PythonOp':
                continue
            for o in n.outputs():
                if o.type().kind() != 'TupleType':
                    continue
                for e in o.type().elements():
                    # In this test, tensor elements in a tuple output 
                    # can't have unknown shapes.
                    assert e.dim()
```
This `assert` throws without my change. Basically, I check the IR before dumping it to file, so we don't need `FileCheck()` here.

Reply:
Hi @gmagogsfm Could you please help review this change? thanks a lot.

Reply:
`FileCheck()` doesn't require dumping IR to a file actually, you can pass a string to it, it is generally nicer to write and read than logics that directly check for a pattern.

Here is an exmaple:
https://github.com/pytorch/pytorch/blob/22e3cc21e535e29deea091adf93e0408139b6aa3/test/test_jit.py#L496

Reply:
Done using `FileCheck`. Thanks.